### ISSUE TITLE
How to normalize the image for custom dataset

### ISSUE THREAD

**Issue by John1231983** [yeephycho/densenet-tensorflow#1]
First of all, I would like to thank your code. It is very useful for me. I have used it and obtain 81% in the flower dataset. 
Now, I will use it for my dataset which includes 10.000 png images in the range 0-255. I have two questions for making/training my dataset

1. I saw in the `build_image_data.py` has png_to_jpeg(image_data), so if my dataset has png file, can it still work for converting tfrecord?
2. Because my image range is from 0-255. I guess we must normalize it to 0-1 for input of densenet network. For that purpose, I think we will change after the [line 56](https://github.com/yeephycho/densenet-tensorflow/blob/master/data_provider/data_provider.py#L56) in the `data provider.py` as

```
image_raw = tf.image.decode_jpeg(image_encoded, channels=3)
image_raw = tf.cast(image_raw , tf.float32) * (1. / 255)

```
Am I right?

**Comment by yeephycho** [comment]
1. It's ok if there's *.png files in your dataset.
2. You can find normalization function in tensorflow, I remember the name is per_image_standardization or something. Besides, whether to do the input normalization is totally up to you, it's not a must for nowadays neural network.

With regards!

**Comment by John1231983** [comment]
Great. I solved it. Very nice function. Could I ask one more question before it closed. I have 18600 images in the training dataset and I wrote it to tfrecord file with output

```
2017-11-22 05:15:57.538392 [thread 0]: Wrote 18600 images to 18600 shards.
2017-11-22 05:15:58.107200: Finished writing all 18600 images in data set.
```

And I have `train.tfrecord` file. However, when I feed it to the network (`python train.py`), just 7440 files are using. Where are other images? 
```
Filling queue with 7440 images before starting to train. This will take some time.
```

Note that, I changed the setting `tf.app.flags.DEFINE_integer("TRAINING_SET_SIZE", 18600, "total image number of training set")`

And

```
f.app.flags.DEFINE_integer('train_shards', 1,
                            'Number of shards in training TFRecord files.')
tf.app.flags.DEFINE_integer('validation_shards', 1,
                            'Number of shards in validation TFRecord files.')

tf.app.flags.DEFINE_integer('num_threads', 1,
                            'Number of threads to preprocess the images.')
```

**Comment by yeephycho** [comment]
Hi Join,

All the training images will be used during training.
The number you saw 7440 is related to the random shuffle function of the training set function provided by tensorflow.

With regards!

**Comment by 07Agarg** [comment]
Hi, 

I am new to tensorflow. I have a confusion regarding these equations.
image_raw = tf.image.decode_jpeg(image_encoded, channels=3)
image_raw = tf.cast(image_raw , tf.float32) * (1. / 255)

These equations seem to be correct to me for an image. But for a batch of images, how to modify these equations? 

I read in the documentation of tensorflow for the function : per_image_standardization. It linearly scales image to have zero mean and unit norm. But I want the image to be in the range of [0, 1] instead of [0-255]. So, cannot use this function.
